File: front-end/src/app/service-models/app/v1/client/s-q-l-report-response-model.ts
import { SQLReportColumns } from './s-q-l-report-columns';

export class SQLReportResponseModel {
    id!: number;
    reportName!: string;
    pageNo!: number;
    pageSize!: number;
    totalCount!: number;
    dataColumns!: Array<SQLReportColumns>;
    reportData: any;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/SQLReportResponseModel.cs
using System.Collections.Generic;
using System.Data;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{  
    public class SQLReportResponseModel
    {
        public int Id { get; set; }
        public string ReportName { get; set; }
        public int PageNo { get; set; }
        public int PageSize { get; set; }
        public int TotalCount { get; set; }
        public List<SQLReportColumns> DataColumns { get; set; }
        public DataTable ReportData { get; set; }
    }
    public class SQLReportColumns
    {
        public int ColumnIndex { get; set; }
        public string ColumnName { get; set; }
    }
}
